src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt

```diff
@@ -216,6 +216,9 @@ class Repository(
           party
         },
       )
+      booking.offenderProgramProfiles.addAll(
+        offenderBuilder.bookingBuilders[bookingIndex].offenderProgramProfiles.map { it.build(booking) },
+      )
```

Comment:
This is a lot simpler than the other builders used above because the course allocation builder knows how to build itself and does the hard work, whereas the above entities do the hard work here. I think I prefer this approach but it will become more apparent where the responsibility should lie once we start to split up Repository.

Reply:
Sounds great - will come back to this example for my DSL after I've got an initial DSL working